keps/sig-auth/2799-reduction-of-secret-based-service-account-token/README.md

```diff
@@ -0,0 +1,463 @@
+
+| Alpha | Beta | GA   |
+| ----- | ---- | ---- |
+| -     | 1.24 | 1.25 |
+
+#### Beta -> GA Graduation
+
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+
+#### Alpha -> Beta Graduation
+
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+- [ ] Document and communicate the available actions that consumers of
+      auto-generated secret-based tokens should take. (migrate to either use
+      tokenrequest or explicitly request secret-based tokens)
+
+#### LegacyServiceAccountTokenTracking
+
+| Alpha | Beta | GA   |
+| ----- | ---- | ---- |
+| 1.24  | 1.25 | 1.26 |
+
+#### Beta -> GA Graduation
+
+- [ ] In use by multiple distributions
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+
+#### Alpha -> Beta Graduation
+
+- [ ] In use by multiple distributions
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+
+#### LegacyServiceAccountTokenCleanUp
+
+| Alpha | Beta | GA   |
+| ----- | ---- | ---- |
+| 1.24  | 1.25 | 1.26 |
+
+#### Beta -> GA Graduation
+
+- [ ] In use by multiple distributions
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+
+#### Alpha -> Beta Graduation
+
+- [ ] In use by multiple distributions
+- [ ] Approved by PRR and scalability
+- [ ] Any known bugs fixed
+- [ ] Tests passing
+
+### Upgrade / Downgrade Strategy
+
+The features can be enabled/disabled via the feature gates in upgrade / downgrade.
+What would be changed is described in "Feature Enablement and Rollback" section.
+
+### Version Skew Strategy
+
+The only touches control plane, so version skew strategy is not applicable.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: LegacyServiceAccountTokenNoAutoGeneration
+  - Components depending on the feature gate: kube-controller-manager
+  - Feature gate name: LegacyServiceAccountTokenTracking
+  - Components depending on the feature gate: kube-apiserver
+  - Feature gate name: LegacyServiceAccountTokenCleanUp:
+  - Components depending on the feature gate: kube-controller-manager
+
+###### Does enabling the feature change any default behavior?
+
+- LegacyServiceAccountTokenNoAutoGeneration: no legacy tokens are auto-generated.
+- LegacyServiceAccountTokenTracking: legacy tokens would have new annotation and a configmap would be created in kube-system.
+- LegacyServiceAccountTokenCleanUp: unused auto-generated legacy tokens will be removed.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+yes for all feature gates.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+- LegacyServiceAccountTokenNoAutoGeneration: the same as enable the feature.
```

Comment:
clarify that this controller will create tokens for serviceaccounts created while the feature was off.

Reply:
done.